### PR TITLE
String compat fixes

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ function validate_mbed(name, handle)
         get_version = Libdl.dlsym(handle, :mbedtls_version_get_string)
         version_ptr = Vector{UInt8}(9)
         ccall(get_version, Void, (Ptr{Void},), version_ptr)
-        version = VersionNumber(bytestring(pointer(version_ptr)))
+        version = VersionNumber(String(pointer(version_ptr)))
```
Comment (malmaud) [inline deps/build.jl]: This isn't correct

Comment (malmaud) [inline deps/build.jl]: Actually it is.

**Comment by tkelman** [comment]
Probably needs minimum Compat version to be bumped to more recent than 0.7.9?

**Comment by malmaud** [comment]
I think as it happens this will work, but I should make another PR soon
that bumps it and switches to using unsafe_string.

**Pull request merged**